### ISSUE TITLE
Allow tasks to receive input and send output from/to different locations

### ISSUE THREAD

**Issue by quinncomendant** [zed-industries/zed#17472]
### Check for existing issues

- [X] Completed

### Describe the feature

_TL;DR: make Zed [Tasks](https://zed.dev/docs/tasks) work like TextMate [Commands](https://macromates.com/manual/en/commands)._

Currently, Tasks have limited input and output:

- Input options: environment variables
- Output options: console

By adding multiple input sources and output destinations, Zed tasks could be even more powerful than the OG TextMate Commands. Here's the input and output options used by TextMate:

- Input options:
  - Environment variables
  - STDIN from:
    - Selection
    - Document
    - Scope
    - Line
    - Word
    - Character
    - Nothing
- Output options:
  - Replace Input
  - Replace Document
  - Insert at Caret
  - Insert After Input
  - Show in New Window
  - Show in Tool Tip
  - Discard
  - Replace Selection

TextMate also has some [Output Type helper functions](https://macromates.com/manual/en/commands#changing_output_type) that allow sending output to different locations during runtime:

- `exit_discard`
- `exit_replace_text`
- `exit_replace_document`
- `exit_insert_text`
- `exit_insert_snippet`
- `exit_show_html`
- `exit_show_tool_tip`
- `exit_create_new_document`

These are used like this:

```bash
if [[ -z $error ]]; then
    exit_show_tool_tip "An error occurred: $error";
else
    exit_create_new_document "$output";
fi
```

### If applicable, add mockups / screenshots to help present your vision of the feature

Here is a screenshot from TextMate showing the UI for configuring commands. Of course, in Zed these input/output options can be configured via the task's JSON.

<img width="1066" alt="SCR-20240905-pwet" src="https://github.com/user-attachments/assets/dbda649b-e626-42a6-96de-caf9ab6e4e2a">

**Comment by hzlimingtang** [comment]
<img width="576" height="243" alt="Image" src="https://github.com/user-attachments/assets/59b578b6-3ee1-4d1f-ada1-d6f3e4ed2bfc" />

**Comment by hzlimingtang** [comment]
<img width="487" height="339" alt="Image" src="https://github.com/user-attachments/assets/5ec792e8-bcb0-46c5-a5a1-ce37884465b1" />

Is it possible to implement the above functionality?
